src/vs/vscode.d.ts

```diff
@@ -5191,10 +5191,35 @@ declare module 'vscode' {
 		 * Register a [TreeDataProvider](#TreeDataProvider) for the view contributed using the extension point `views`.
 		 * @param viewId Id of the view contributed using the extension point `views`.
 		 * @param treeDataProvider A [TreeDataProvider](#TreeDataProvider) that provides tree data for the view
+		 * @deprecated Use [createTreeView](#window.createTreeView) instead.
```

Comment:
Use the strikeout styles we use in other places. Also, we could consider not deprecating this but explaining the pros and cons...